code/game/gamemodes/malfunction/Malf_Modules.dm

```diff
@@ -36,6 +36,10 @@
 		if(initial(uses) > 1) //no need to tell 'em if it was one-use anyway!
 			to_chat(owner, "<span class='warning'>[name] has run out of uses!</span>")
 		qdel(src)
+		if(istype(src, /datum/action/innate/ai/overload_machine))

```

Comment:
Never do this. OOP Principles means that object behavior should be attached to the object.

This means that instead of every action checking whether it is overload machine, it should be overload machine doing this adjustment instead.

Now I'm not too sure if this is necessary as I don't see this system being used to add actions in the old PR. Cannot comment on that.

Reply:
Yeah this is something I added actually. I felt like it didn't make sense for the action button to be removed when amount of charges reaches 0 but then have the verb not get removed.

I see what you mean about having the overload/override procs handle this.